### PR TITLE
Extend the CoC to cover all interactions

### PR DESCRIPTION
**Pull request by RichardLitt**
- [n/a ] Code compiles correctly
- [n/a ] All tests are passing
- [n/a ] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [n/a ] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

Previously, the guide implied it only applied to Pull Requests. See https://github.com/ethereum/solidity/issues/4445#issuecomment-418159492.

### CODE REVIEW

**Review by axic — DISMISSED** [review]
I think it is fine.

**Review by chriseth — DISMISSED** [review]

File: docs/contributing.rst [hunk 1]
```diff
@@ -19,6 +19,11 @@ In particular, we need help in the following areas:
   `up-for-grabs <https://github.com/ethereum/solidity/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs>`_ which are
   meant as introductory issues for external contributors.
 
+Please note that this project is released with a `Contributor Code of Conduct
+  <https://raw.githubusercontent.com/ethereum/solidity/develop/CODE_OF_CONDUCT.md>`_.
```
Comment (chriseth) [inline docs/contributing.rst]: I think you cannot indent here, at least it does not compile.

Comment (RichardLitt) [reply docs/contributing.rst]: Should be fixed.

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
We might want to link the code of conduct from the channel titles in the two gitter channels to make people aware.

**Comment by chriseth** [comment]
I added a welcome message to the gitter rooms.

**Comment by RichardLitt** [comment]
Awesome. Thanks, @chriseth!

**Pull request merged**